cmd/root.go

```diff
@@ -251,11 +266,30 @@ func initConfig() {
 		logger.WithError(err).Fatal("Error creating kubernetes client")
 	}
 
-	rm, err = manager.NewResourceManager(k8sClient)
+	// Create a shared informer factory for Kubernetes pods in the current namespace (if specified) and scheduled to the current node.
+	podInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(k8sClient, kubeSharedInformerFactoryResync, kubeinformers.WithNamespace(kubeNamespace), kubeinformers.WithTweakListOptions(func(options *metav1.ListOptions) {
+		options.FieldSelector = fields.OneTermEqualSelector("spec.nodeName", nodeName).String()
+	}))
+	// Create a pod informer so we can pass its lister to the resource manager.
+	podInformer = podInformerFactory.Core().V1().Pods()
+
+	// Create another shared informer factory for Kubernetes secrets and configmaps (not subject to any selectors).
+	scmInformerFactory := kubeinformers.NewSharedInformerFactoryWithOptions(k8sClient, kubeSharedInformerFactoryResync)
```

Comment:
That's because listers obtained from the `podInformerFactory` are all subject to what is specified in `WithTweakListOptions`. Have in mind that [the method you are using in your PR is deprecated](https://github.com/kubernetes/client-go/blob/master/informers/factory.go#L99-L100) in favor of what I'm adopting here.

If we create a secret/configmap informer from `podInformerFactory` - so that we could grab its lister - we would see error messages indicating that `spec.nodeName` is not a valid selector for secrets/configmaps. On the other hand, we must constrain the `podInformerFactory` with said selector, since we are only interested in pods (and in particular pod events) that have been scheduled to the current node. Hence, we need two separate factories.

Reply:
The deprecation only seems to apply with the method without options, but it doesn't seem like it's deprecated to create an informer from another?
The shared informer factory seems to lose value if you can't have listers with different list options applied coming from the same informer.